go/consensus/tendermint/apps/roothash/transactions.go

```diff
@@ -140,20 +141,25 @@ func (app *rootHashApplication) executorCommit(
 		app.tryFinalizeExecute(ctx, rtState, pool, false)
 	}
 
+	// Update runtime state.
+	if err = state.SetRuntimeState(ctx, rtState); err != nil {
+		return fmt.Errorf("failed to set runtime state: %w", err)
```

Comment:
> how might adding commitments fail? like if one has an invalid signature, then the rest are also not stored? I guess that doesn't matter much in this case because they're all coming from the same transaction sender

Yeah invalid signature, not based on correct block, not in committee, etc. So currently compute nodes only send a single commitment (their own) so this is not a problem in practice. If we had commit aggregators we would probably want to just skip over invalid commits instead of returning an error.

Reply:
ok yeah. aggregators ought to check too, so things shouldn't get too crazy even if we forget about this